### PR TITLE
BXMSDOC-4514 Document Migration of spaces and repositories configurations from system.git to each space config.git

### PR DESCRIPTION
**Pull request by emmurphy1**
Details in jira:
[BXMSDOC-4514](https://issues.jboss.org/browse/BXMSDOC-4514)

### CODE REVIEW

**Review by tomasdavidorg — APPROVED** [review]

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+= Using the migration tool on a `.niogit` directory
+
+The `.niogit` directory of your {CENTRAL} distribution contains all {CENTRAL} data. You can use the {CENTRAL} migration tool on your `.niogit` directory to prepare project data and system configurations for migration to {PRODUCT} 7.5.
+
+
+.Prerequisites
+* {PRODUCT} 7.5 is installed.
+* {PRODUCT} 7.4 in installed and {CENTRAL} is not running on {PRODUCT} 7.4.
+* Java 8 is installed and is available on the path where the project migration tool will be installed.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: JVM? JDK?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,217 @@
+[id='migration-tool-niogit-7.4-proc_{context}']
+
+= Using the migration tool on a `.niogit` directory
+
+The `.niogit` directory of your {CENTRAL} distribution contains all {CENTRAL} data. You can use the {CENTRAL} migration tool on your `.niogit` directory to prepare project data and system configurations for migration to {PRODUCT} 7.5.
+
+
+.Prerequisites
+* {PRODUCT} 7.5 is installed.
+* {PRODUCT} 7.4 in installed and {CENTRAL} is not running on {PRODUCT} 7.4.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: 7.4 and 7.5 on the same machine, in different directories?

Comment (mramendi) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Also you have to have 7.4 to migrate, with 7.2/7.3 this won't work? If so, I suggest pointing it out explicitly.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+----
+$ ~/${PRODUCT_INIT_CAP}_7.4_HOME/bin/.niogit
+----
+
+[source,subs="attributes+"]
+----
+$ ~/${PRODUCT_INIT_CAP}_7.5_HOME/bin/.niogit
+----
+--
+. In your command terminal, copy the `.niogit` directory from your current {CENTRAL} distribution for {PRODUCT} 7.4 to the `bin` folder of the new {PRODUCT} 7.5 installation directory:
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:34]: That means overwriting the 7.5 directory that one located in the previous step?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 2 above)
Comment (mramendi) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Also no version variables, this doc forever remains for 7.4 and 7.5?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+[source,subs="attributes+"]
+----
+cp -r /${PRODUCT_INIT_CAP}_7.4_HOME/bin/.niogit  /${PRODUCT_INIT_CAP}_7.5_HOME/bin/
+----
+
+The copied `.niogit` folder is the directory that the migration tool will prepare for {PRODUCT} 7.5.
+. The original `.niogit` folder remains in your {PRODUCT} 7.4 installation and will not be modified by the migration tool in this procedure.
+--
+. In the {PRODUCT} 7.5. `bin` directory, delete the `.index` directory. You must delete this folder because {PRODUCT} 7.5 comes with certain updates that are not compatible with {PRODUCT} {PRODUCT_VERSION}. The `bin` directory will be re-indexed after the migration when you start {PRODUCT} 7.5.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: With the current values, the phrase here will read "because RHPAM 7.5 comes with certain updates that are not compatible with RHPAM 7.5", so you probably meant something else?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  "repositories-add":[],
+  "repositories-remove":[],
+  "repositories-update-urls":[],
+  "pluginRepositories-add":[],
+  "pluginRepositories-remove":[],
+  "pluginRepositories-update-urls":[]
+}
+----
+--
+. Navigate to the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Software Downloads] page in the Red Hat Customer Portal (login required), and select the product and version from the drop-down options:
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:157]: We have this URL in the artifacts somewhere? Not sure, perhaps check with Stetson

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
++
+--
+On Linux or UNIX-based systems:
+[source,subs="attributes+"]
+----
+$ cd $INSTALL_DIR/{PRODUCT_INIT}-7.5-migration-tool/bin
+$ ./migration-tool.sh -t ${PRODUCT_INIT_CAP}_NIOGIT_DIR
+----
+
+On Windows:
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: "On Windows systems" if you want to follow IBM convention (and I'm writing this as a former IBM information developer who fixed this very thing in a number of peer reviews back there)

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  "pluginRepositories-remove":[],
+  "pluginRepositories-update-urls":[]
+}
+----
+--
+. Navigate to the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Software Downloads] page in the Red Hat Customer Portal (login required), and select the product and version from the drop-down options:
+* *Product:* {PRODUCT_SHORT}
+* *Version:* 7.5
+
+. Download *{PRODUCT} 7.5 Add-Ons* and extract the downloaded `{PRODUCT_INIT}-7.5.0-add-ons.zip` file to a temporary directory.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:161]: again no version variables? This might become a maintenance issue?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+----
+endif::DM[]
+ifdef::PAM[]
+----
+$ cd $INSTALL_DIR\rhpam-7.5-migration-tool\bin
+$ migration-tool.bat -t ${PRODUCT_INIT_CAP}_NIOGIT_DIR
+----
+endif::PAM[]
+
+In the command prompt that appears, the following options are displayed:
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:188]: passive voice and all that

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+
+* *Project structure migration*: Migrates the {PRODUCT} 7.4 project repository structure to the new project-oriented structure used in {PRODUCT} {PRODUCT_VERSION}.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: You used 7.5 before and {PRODUCT_VERSION} now. I would suggest consistency one way or the other.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+* *System configuration directory structure migration*: Migrates the `system.git` repository structure used in {PRODUCT} 7.4 to the new structure used in {PRODUCT} 7.5. This migration option requires the project structure migration to be executed first.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:191]: I would rephrase to avoid passive voice, something like "You must complete the project structure migration before selecting this option"

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+ifdef::PAM[]
+* *Forms migration*: Migrates forms created in the {PRODUCT_OLD} forms designer to the new forms designer. This migration option requires the project structure migration and system configuration directory structure migration to be executed first.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Again I'd rephrase to avoid passive voice

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+to be executed first.
+* *All:* Runs all migration options in sequence.
+* *Exit:* Exits the migration tool.
+--
+. Select the option to run *All* migrations in sequence.
++
+NOTE: If you prefer to run one migration option at a time, select and run the first individual migration option. After the tool runs, re-run the {CENTRAL} migration tool and select the next individual migration option in the sequence.
++
+
+. Enter `yes` each time you are prompted to run a specific migration option.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:211]: Does it really require all three letters "yes"?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,217 @@
+[id='migration-tool-niogit-7.4-proc_{context}']
+
+= Using the migration tool on a `.niogit` directory
+
+The `.niogit` directory of your {CENTRAL} distribution contains all {CENTRAL} data. You can use the {CENTRAL} migration tool on your `.niogit` directory to prepare project data and system configurations for migration to {PRODUCT} 7.5.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:5]: Probably mention something about not running BC 7.5 before doing this migration stuff?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
++
+NOTE: If you prefer to run one migration option at a time, select and run the first individual migration option. After the tool runs, re-run the {CENTRAL} migration tool and select the next individual migration option in the sequence.
++
+
+. Enter `yes` each time you are prompted to run a specific migration option.
++
+For the POMs migration option, if you want to include a path to an external `pom-migration.json` file that you created previously, enter `yes` when prompted and enter the path.
++
+. After the tool finishes running, enter the option to *Exit* the migration tool.
++
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:216]: Should not this and the next one be under "Results" and not a step?

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.x-proc.adoc [hunk 15]
```diff
@@ -166,42 +253,119 @@ Example `pom-migration.json` file with `dependencies` only:
...
+ifeval::["{context}" == "non-exec-server"]
+{PRODUCT_VERSION}
+endif::[]
+ifeval::["{context}" == "migrate-7.0-to-7.x"]
+{PRODUCT_VERSION}
+endif::[]
+ifeval::["{context}" == "patching-upgrading"]
+7.5
+endif::[]
+. This migration option requires the project structure migration to be executed first.
```
Comment (mramendi) [inline doc-content/enterprise-only/migration/migration-tool-niogit-7.x-proc.adoc:354]: Formatted as a step?..

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 5 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:157]: Not that I am aware of. This page is maintained by Martin Weiler in Support.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 2 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Yes. That is correct. This update is for the 7.4 patching doc hence the static values. And this section is specifically about upgrading from 7.4 to 7.5. I think it's clear in the rendered docs.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 4 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Again, this is clear in the rendered doc:
In the Red Hat Process Automation Manager 7.5. bin directory, delete the .index directory. You must delete this folder because Red Hat Process Automation Manager 7.5 comes with certain updates that are not compatible with Red Hat Process Automation Manager 7.4. The bin directory will be re-indexed after the migration when you start Red Hat Process Automation Manager 7.5.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 7 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:161]: This is common in the patching docs - each is unique and usually contains static values.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 4 above)
Comment (mramendi) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: After a rebase, however, {PRODUCT_VERSION} will become 7.5. I would suggest hardcoding 7.4 here.

Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Well, this is for 7.4 only, but I have hardcoded in two places.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 14 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:216]: I don't think so. They are both actions.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 1 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Tomas, I changed to JDK 8. Please confirm.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 9 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: Removed variable.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 3 above)
Comment (tomasdavidorg) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:34]: There should not be any .niogit directory if the 7.5 was not started before.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 1 above)
Comment (tomasdavidorg) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc]: OK to me.

File: doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc (hunk 12 above)
Comment (tomasdavidorg) [reply doc-content/enterprise-only/migration/migration-tool-niogit-7.4-proc.adoc:211]: Yes. After every migration option, you have to confirm by "yes". There is also a parameter `-b` which run all the steps without asking .
